plugin/src/main/java/org/opensearch/ml/indices/MLIndicesHandler.java

```diff
@@ -97,12 +98,19 @@ public void initMLIndexIfAbsent(MLIndex index, ActionListener<Boolean> listener)
                                 .putMapping(
                                     new PutMappingRequest().indices(indexName).source(mapping, XContentType.JSON),
                                     ActionListener.wrap(response -> {
-                                        if (response.isAcknowledged()) {
-                                            indexMappingUpdated.get(indexName).set(true);
-                                            internalListener.onResponse(true);
-                                        } else {
-                                            internalListener.onFailure(new MLException("Failed to update index: " + indexName));
-                                        }
+                                        UpdateSettingsRequest updateSettingRequest = new UpdateSettingsRequest();
+                                        updateSettingRequest.indices(indexName).settings(indexSettings);
+                                        client
+                                            .admin()
+                                            .indices()
+                                            .updateSettings(updateSettingRequest, ActionListener.wrap(updateResponse -> {
+                                                if (response.isAcknowledged()) {
+                                                    indexMappingUpdated.get(indexName).set(true);
+                                                    internalListener.onResponse(true);
+                                                } else {
+                                                    internalListener.onFailure(new MLException("Failed to update index: " + indexName));
```

Comment:
Tune the error message 
```suggestion
                                                    internalListener.onFailure(new MLException("Failed to update index setting: " + indexName));
```